### PR TITLE
[SOAR-10513] Palo Alto Cortex XDR - Added typing to Get XQL Query Results action | Added new trigger Get Query Results

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Added typing to Get XQL Query Results action
  - Added new trigger Get Query Results

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/triggers/get_query_results/trigger.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        interval = abs(params.get(Input.FREQUENCY, 5))
+
+        end_time = Util.now_ms() - interval * 1000
+
+        self.logger.info("Initializing Get Query Results trigger for the Palo Alto Cortex XDR plugin.")
+
+        while True:
+            start_time = Util.now_ms()
+            execution_id = self.connection.xdr_api.start_xql_query(query, tenants, end_time, start_time)
+            query_results = insightconnect_plugin_runtime.helper.clean(
```
Comment (jkaluza-rapid7) [inline plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/triggers/get_query_results/trigger.py:31]: This method does not clean nested dictionaries well. Do results contain nested dicts?

Comment (igorski-r7) [reply plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/triggers/get_query_results/trigger.py:31]: I think there is one `dict` in it, but nothing much deeper 😃

File: plugins/palo_alto_cortex_xdr/help.md [hunk 2]
```diff
@@ -661,45 +689,248 @@ Example output:
...
+|Insertion timestamp|integer|False|Integer representing insertion timestamp as epoch in ms|
+
 
 ## Troubleshooting
 
 * Isolate Endpoint fails with 500 error - This will happen if an isolation action (Isolate or Unisolate) is in progress on the selected endpoint. Wait a few minutes and try again. 
 
 # Version History
 
+* 3.0.0 - Add types to `Get XQL Query Results` | Add new trigger `Get Query Results`
```
Comment (blaxminarayan-r7) [inline plugins/palo_alto_cortex_xdr/help.md:933]: I see that some types have been changed in incident which is causing backward compatibiilty. Are they necessary. If so can you please add the fields in the version history.

**Review by blaxminarayan-r7 — CHANGES_REQUESTED** [review]

File: plugins/palo_alto_cortex_xdr/help.md [hunk 3]
```diff
@@ -661,45 +689,248 @@ Example output:
...
+|Insertion timestamp|integer|False|Integer representing insertion timestamp as epoch in ms|
+
 
 ## Troubleshooting
 
 * Isolate Endpoint fails with 500 error - This will happen if an isolation action (Isolate or Unisolate) is in progress on the selected endpoint. Wait a few minutes and try again. 
 
 # Version History
 
+* 3.0.0 - Add types `xql_query_result, xql_query_result_data` to `Get XQL Query Results` | Add new trigger `Get Query Results`
```
[the remark this answers is not in the document]
Comment (igorski-r7) [reply plugins/palo_alto_cortex_xdr/help.md:933]: Well, the only change here is just added typing for the response fields and new trigger. I know it should be versioned `2.3.0`, but the plugin validators are failing due to a `Version Increment Validator` which it suggests being `3.0.0`

```
Validator "Version Increment Validator" failed! 
        Cause: Type results changed in type reply without a major version increment. Please change the plugin version to 3.0.0
```

**Review by blaxminarayan-r7 — APPROVED** [review]

**Pull request merged**